Utils/hdinsight-node-common/src/com/microsoft/azure/sparkserverless/serverexplore/SparkServerlessClusterUpdateCtrlProvider.java

```diff
@@ -0,0 +1,71 @@
+package com.microsoft.azure.sparkserverless.serverexplore;
+
+import com.microsoft.azure.hdinsight.common.logger.ILogger;
+import com.microsoft.azure.hdinsight.common.mvc.IdeSchedulers;
+import com.microsoft.azure.hdinsight.common.mvc.SettableControl;
+import com.microsoft.azure.hdinsight.sdk.common.azure.serverless.AzureSparkServerlessCluster;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import org.apache.commons.lang3.StringUtils;
+import rx.Observable;
+
+public class SparkServerlessClusterUpdateCtrlProvider implements ILogger {
+    @NotNull
+    private SettableControl<SparkServerlessClusterProvisionSettingsModel> controllableView;
+    @NotNull
+    private IdeSchedulers ideSchedulers;
+    @NotNull
+    private AzureSparkServerlessCluster cluster;
+
+    public SparkServerlessClusterUpdateCtrlProvider(
+            @NotNull SettableControl<SparkServerlessClusterProvisionSettingsModel> controllableView,
+            @NotNull IdeSchedulers ideSchedulers,
+            @NotNull AzureSparkServerlessCluster cluster) {
+        this.controllableView = controllableView;
+        this.ideSchedulers = ideSchedulers;
+        this.cluster = cluster;
+    }
+
+    public Observable<SparkServerlessClusterProvisionSettingsModel> initialize() {
+        return cluster.get()
+                .observeOn(ideSchedulers.processBarVisibleAsync("Updating cluster status..."))
+                .map(clusterUpdated -> {
+                    SparkServerlessClusterProvisionSettingsModel toUpdate =
+                            new SparkServerlessClusterProvisionSettingsModel();
+                    controllableView.getData(toUpdate);
+                    return toUpdate
+                            .setClusterName(clusterUpdated.getName())
```

Comment:
I suppose needn't to update these.

Reply:
By default the information in this dialog is the same as it in provision dialog since `SparkServerlessClusterUpdateDialog` extends from `SparkServerlessProvisionDialog`. However, the provision dialog shows the default value we set rather than the real value of a specific cluster.

---
In reply to: [197368629](https://github.com/Microsoft/azure-tools-for-java/pull/1734#discussion_r197368629) [](ancestors = 197368629)